pkg/timeout/handler.go

```diff
@@ -186,7 +198,7 @@ func (t *TimeoutSet) checkPipelineRunTimeouts(namespace string, pipelineclientse
 
 // CheckTimeouts function iterates through a given namespace or all namespaces
 // (if empty string) and calls corresponding taskrun/pipelinerun timeout functions
-func (t *TimeoutSet) CheckTimeouts(namespace string, kubeclientset kubernetes.Interface, pipelineclientset clientset.Interface) {
+func (t *Handler) CheckTimeouts(namespace string, kubeclientset kubernetes.Interface, pipelineclientset clientset.Interface) {
```

Comment:
its interesting that we are scrapping through all possible namespaces (if not specified) and checking all `TaskRuns` and `PipelineRuns` in those namespaces or at least in one specified namespace 😲

Reply:
And the same check `CheckTimeouts` is done in TaskRun controller and PipelineRun controller 🤔

Reply:
yeah!! the second means we're probably doing this twice as much as we need to 😅 

and it sounds like we probably don't need to be doing it at all!! https://github.com/tektoncd/pipeline/issues/2905#issuecomment-663717912